modules/standard/Map.chpl

```diff
@@ -469,7 +461,7 @@ module Map {
                `false` otherwise.
      :rtype: bool
     */
-    proc add(in k: keyType, in v: valType): bool {
+    proc add(in k: keyType, in v: valType): bool lifetime this < v {
```

Comment:
I think we should also have a check for `this < k` as well.

Reply:
I agree that that makes sense, but would like to do it as a separate PR if that's OK with you.

Reply:
Is the idea there that you'll add checks for borrowed keys at that time as well?

Reply:
Yes, I think adding tests would need to be a part of that if not earlier.